bull/trove-queue/consumer.js

```diff
@@ -81,18 +89,25 @@ TroveQueue.process((job, done) => {
                     level: "error",
                     message: `IA Failure Trove ${error}`,
                   });
+                  if (isEmailNotification === "true") {
+                    EmailProducer(userName, name, trueURI, false);
+                  }
                   done(new Error(error));
                 } else {
                   logger.log({
                     level: "error",
                     message: `IA Failure Trove ${body}`,
                   });
+                  if (isEmailNotification === "true") {
+                    EmailProducer(userName, name, trueURI, false);
+                  }
                   done(new Error(body));
                 }
-                //EmailProducer(job.data.details.email, name, trueURI, false);
               } else {
+                if (isEmailNotification === "true") {
+                  EmailProducer(userName, name, trueURI, true);
```

Comment:
No need of repeating it. Keep it outside the if else block and call the function once in `if (error || response.statusCode != 200) {` block.